File: LeetCode/Easy/Problem 704/p704Recursive.c
int binSearch(int* nums, int target, int start, int end) {
    if (start > end) {
        return -1;
    }
    else if (start == end) {
        if (nums[start] == target) {
            return start;
        }
        else {
            return -1;
        }
    }
    
    int mid = (start + end) / 2;
    if (nums[mid] == target) {
        return mid;
    }
    else if (nums[mid] < target) {
        return binSearch(nums, target, mid + 1, end);
    }
    else if (nums[mid] > target) {
        return binSearch(nums, target, start, mid - 1);
    }
    
    return 0; //Prevents warning message for no return even though this will never be reached
}

int search(int* nums, int numsSize, int target){
    return binSearch(nums, target, 0, numsSize - 1);
}

File: LeetCode/Easy/Problem 704/p704Recursive.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        return self.binSearch(nums, target, 0, len(nums) - 1)
        
    def binSearch(self, nums, target, start, end):
        if start > end:
            return -1
        elif start == end: 
            if nums[start] == target:
                return start
            else:
                return -1
        
            
        mid = (start + end) // 2
        if nums[mid] == target:
            return mid
        elif nums[mid] < target:
            return self.binSearch(nums, target, mid + 1, end)
        elif nums[mid] > target:
            return self.binSearch(nums, target, start, mid - 1)
        